src/System Application/App/AI/src/Azure OpenAI/AzureOpenAIImpl.Codeunit.al

```diff
@@ -474,12 +458,17 @@ codeunit 7772 "Azure OpenAI Impl"
     end;
 
     [NonDebuggable]
-    internal procedure GetMetaprompt() Metaprompt: SecretText;
+    internal procedure GetTextMetaprompt() Metaprompt: SecretText;
     var
         AzureKeyVault: Codeunit "Azure Key Vault";
+        EnvironmentInformation: Codeunit "Environment Information";
         KVSecret: Text;
     begin
-        if AzureKeyVault.GetAzureKeyVaultSecret('AOAI-Metaprompt', KVSecret) then;
+        if EnvironmentInformation.IsSaaSInfrastructure() then
```

Comment:
You could do this instead:
if not IsSaaS() then exit('');

Would reduce nesting and make it very clear what this function does.